pkg/instance/build.go

```diff
@@ -226,7 +225,11 @@ func getImageRegistry(imageName string) (string, error) {
 
 // getBuildDir returns the build directory for the instance
 func (b *build) getBuildDir() string {
-	return filepath.Join(buildDirBase, b.instance.name)
+	tmpDir, err := os.MkdirTemp("", "knuu-build-*")
```

Comment:
why the `*`?

Reply:
you mean why `b *build` ? 
if so, we need to pass it by reference because it is more performant than not using it which copies the entire object. Also we need it as there is a `*sync.Map` in it, and this way it works properly.

Reply:
```
func os.MkdirTemp(dir string, pattern string) (string, error)
MkdirTemp creates a new temporary directory in the directory dir and returns the pathname of the new directory. The new directory's name is generated by adding a random string to the end of pattern. If pattern includes a "*", the random string replaces the last "*" instead. The directory is created with mode 0o700 (before umask). If dir is the empty string, MkdirTemp uses the default directory for temporary files, as returned by TempDir. Multiple programs or goroutines calling MkdirTemp simultaneously will not choose the same directory. It is the caller's responsibility to remove the directory when it is no longer needed.
```